commit 3114bcb097ba
Author: Bryce Van Dyk <bvandyk@mozilla.com>
Date:   Wed Feb 28 16:34:05 2018 -0500

    Bug 868797 - Update libnestegg to be more tolerant of blocks with negative timecodes. r=kinetik
    
    Bump nestegg to commit 89ed0daf2edccb25f744e5faff88b8b4684adceb. This brings
    across tolerance of blocks with negative timecodes. Instead of rejecting these
    the timecodes are now set to 0.
    
    Also brings across a change to appease clang in ne_read_block_additions by
    adding an explicit assignment to data_size.
    
    MozReview-Commit-ID: 7J8YPUUwSBp
    
    --HG--
    extra : rebase_source : f55bd987465baf21f383095b60e9148349936fef
---
 media/libnestegg/README_MOZILLA | 2 +-
 media/libnestegg/src/nestegg.c  | 8 ++++++--
 2 files changed, 7 insertions(+), 3 deletions(-)

diff --git media/libnestegg/README_MOZILLA media/libnestegg/README_MOZILLA
index 979b4e936dfe..632bbfaedbe0 100644
--- media/libnestegg/README_MOZILLA
+++ media/libnestegg/README_MOZILLA
@@ -5,4 +5,4 @@ Makefile.in build files for the Mozilla build system.
 
 The nestegg git repository is: https://github.com/kinetiknz/nestegg
 
-The git commit ID used was 017f480db78bcdc607e6dba36cef806b793edf2c.
+The git commit ID used was 89ed0daf2edccb25f744e5faff88b8b4684adceb.
diff --git media/libnestegg/src/nestegg.c media/libnestegg/src/nestegg.c
index 133732e481bc..c8939f634210 100644
--- media/libnestegg/src/nestegg.c
+++ media/libnestegg/src/nestegg.c
@@ -1550,8 +1550,11 @@ ne_read_block(nestegg * ctx, uint64_t block_id, uint64_t block_size, nestegg_pac
   cluster_tc = ctx->cluster_timecode;
 
   abs_timecode = timecode + cluster_tc;
-  if (abs_timecode < 0)
-    return -1;
+  if (abs_timecode < 0) {
+      /* Ignore the spec and negative timestamps */
+      ctx->log(ctx, NESTEGG_LOG_WARNING, "ignoring negative timecode: %lld", abs_timecode);
+      abs_timecode = 0;
+  }
 
   pkt = ne_alloc(sizeof(*pkt));
   if (!pkt)
@@ -1686,6 +1689,7 @@ ne_read_block_additions(nestegg * ctx, uint64_t block_size, struct block_additio
     add_id = 1;
     data = NULL;
     has_data = 0;
+    data_size = 0;
     r = ne_read_element(ctx, &id, &size);
     if (r != 1)
       return r;
